LearnositySDK/Request/DataApi.cs

```diff
@@ -160,7 +160,7 @@ private JsonObject handleRequestRecursive(string url, JsonObject securityPacket,
                     }
                 }
             }
-            while(Tools.array_key_exists("next", meta));
+            while(Tools.array_key_exists("next", meta) && !Tools.empty(data.getJsonObject("data")));
```

Comment:
Hi @karoltarasiuk, sorry for the delay in getting back to you. When you mentioned a refactor to align with PHP SDK. Are you sure about that? Or should we refactor PHP SDK and remove the if from line 79 (https://github.com/Learnosity/learnosity-sdk-php/blob/master/src/LearnositySdk/Request/DataApi.php#L79) and add that condition to the while? Let me know your thoughts. Cheers!